src/hazmat/primecount.rs

```diff
@@ -0,0 +1,117 @@
+use crypto_bigint::{Uint, U256};
+
+/// Estimate the number of primes smaller than x using the asymptotic expansion of Li(x) with three terms. 𝜋(𝑥)/Li(x)
+/// converges to 1 from below, which means that the estimation of number of primes for the ranges relevant here is a
+/// slight overestimate.
```

Comment:
`Li(x) > pi(x)`, but you're using `approx(Li(x)) < Li(x)`. Is it always the case that `approx(Li(x)) > pi(x)`?

Reply:
Also `ln(x)` that you use to appoximate `Li(x)` is itself approximated, which adds to the error.

Reply:
> Is it always the case that approx(Li(x)) > pi(x)?

I don't know. For the ranges considered, it's true that Li(x) > pi(x), but it's not true in the general case; indeed Li(x) - pi(x) changes sign infinitely many times apparently. How the approximation behaves compared to Li(x) I simply do not know.

Reply:
> but it's not true in the general case 

Yeah, the first known Skewes's number is huge, so we shouldn't worry about that. For our use we can assume `Li(x) > pi(x)`. But I think it would be useful if we could make an assertion that the approximation we return is always an upper bound compared to a weaker statement that it's just somewhere close.

Reply:
I'll ask an AI!

Reply:
Ha, it actually delivered, pointing out that the dominant error here is from the flooring, so approx_ln_x <= actual_ln_x. This means that approx_pi_x actually overestimates Li(x), given that all terms in the expansion divides by powers of ln_x).

So in conclusion:  it is always the case that approx(Li(x)) > pi(x) (for these ranges).

Reply:
Yeah, but you're omitting terms in the expansion of `Li(x)` which makes it smaller?

Reply:
4!/ln(x)^4 is tiny, ~10^-11 for x = 2^1023, so given approx_Li overestimates by ~0.1% we're ok.